autovalet_goal_generation/scripts/transfer_lane.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+'''
+Script runs an approximate time policy message filter to catch
+depthmap and segmentation maps at the same time and produces a
+segmented lane cloud and publishes it
+
+Author  : Subramanian Krishnan (subramak@andrew.cmu.edu)
+Date    : 09 September 2020
+
+Changelog:
+    subbu - 9/07 - Initial commit
+    subbu - 9/07 - Lane seg overlaid on depthmap
+    eva/subbu - 9/12 - Fixed fov
+    subbu - 9/14 - lane line extraction from segmap
+    subbu - 9/18 - Depth registration added
+'''
+import numpy as np
+import rospy
+import sensor_msgs.point_cloud2 as pcl2
+import cv2
+
+from sensor_msgs.msg import Image, CameraInfo, PointCloud2
+from std_msgs.msg import Header
+from cv_bridge import CvBridge, CvBridgeError
+from message_filters import ApproximateTimeSynchronizer, Subscriber
+
+class transfer:
+
+  def __init__(self, segmap_topic, depthmap_topic, cameraInfo_topic, laneCloud_topic):
+    self.bridge = CvBridge()
+    self.depth_sub  = Subscriber(depthmap_topic, Image)
+    self.segmap_sub = Subscriber(segmap_topic, Image)
+
+    # Read camera info by looking up only one message
+    self.camera = rospy.wait_for_message(cameraInfo_topic, CameraInfo)
+    self.lane_cloud_pub = rospy.Publisher(laneCloud_topic, PointCloud2, queue_size=1)
+
+    # Looking to synchronize both the topics within a 1/10th of a second
+    self.ats = ApproximateTimeSynchronizer([self.depth_sub, self.segmap_sub], queue_size=2, slop=0.1)
+    self.ats.registerCallback(self.callback)
+
+    # Define the camera matrix individual values for projection
+    self.fx     = self.camera.K[0]
+    self.fy     = self.camera.K[4]
+    self.cx     = self.camera.K[2]
+    self.cy     = self.camera.K[5]
+
+    # Define kernel for dilation of lanes
+    self.dilation_kernel = np.ones((2,2),np.uint8)
+
+  def callback(self,depthmap, segmap):
+    # process image
+    depth_image = self.bridge.imgmsg_to_cv2(depthmap, desired_encoding="passthrough")
+    segmap_image = self.bridge.imgmsg_to_cv2(segmap, desired_encoding="passthrough")
+
+    # Close the segmap to remove small holes in the segmented lane
+    segmap_processed = cv2.morphologyEx(segmap_image, cv2.MORPH_CLOSE, self.dilation_kernel)
+    # Generate new mask with lane boundary
+    lane_lines = cv2.Canny(segmap_processed, 100, 200)
+    # Perform morphological opening to increase width of lane line
+    lane_lines = cv2.dilate(lane_lines,self.dilation_kernel,iterations = 1)
+
+    # Overlay mask
+    lane_depth_image = cv2.bitwise_and(depth_image, depth_image, mask = lane_lines/255)
```

Comment:
is "depth_image" both src and destination? wondering why it comes up twice in parameters

Reply:
I used this as a reference .. didn't dig too deep there
https://docs.opencv.org/master/d0/d86/tutorial_py_image_arithmetics.html